src/parser/mod.rs

```diff
@@ -105,22 +104,21 @@ impl<'a> WldDoc<'a> {
 
     /// Iterate over all mesh fragments in the wld file.
     pub(super) fn meshes(&self) -> impl Iterator<Item = (Option<&str>, MeshFragment)> + '_ {
-        self.fragment_iter::<MeshFragment>(MESH_FRAGMENT_ID)
+        self.fragment_iter::<MeshFragment>()
     }
 
     /// Iterate over all material fragments in the wld file.
     pub(super) fn materials(&self) -> impl Iterator<Item = (Option<&str>, MaterialFragment)> + '_ {
-        self.fragment_iter::<MaterialFragment>(MATERIAL_FRAGMENT_ID)
+        self.fragment_iter::<MaterialFragment>()
     }
 
-    fn fragment_iter<T: Fragment<T = T> + Debug>(
+    pub fn fragment_iter<T: Fragment<T = T> + Debug>(
```

Comment:
Adding the `TYPE_ID` to the fragment trait allows you to find a specific fragment given _only_ the type. That means you're no longer required to pass in the fragment type `u32` here. Before it was a little bit odd that you needed both. And what would happen if you passed in a type and `u32` that didn't match? Unclear 😆.